include/random_walks/gaussian_hamiltonian_monte_carlo_exact_walk.hpp

```diff
@@ -218,12 +226,47 @@ private :
             }
             _lambda_prev = pbpair.first;
             update_position(_p, _v, _lambda_prev, _omega);
+            nudge_in(P_normalized, _p);
             T -= _lambda_prev;
             P.compute_reflection(_v, _p, pbpair.second);
             it++;
         }
     }
 
+    template
+    <
+        typename GenericPolytope
+    >
+    inline void nudge_in(GenericPolytope& P, Point& p, NT tol=NT(0))
+    {
+        MT A = P.get_mat();
+        VT b = P.get_vec();
+
+        int m = A.rows();
+        const NT* b_data = b.data();
+
+        for (int i = 0; i < m; i++) {
+
+            NT dist = *b_data - A.row(i) * p.getCoefficients();
```

Comment:
You can do it as follows (because it is more efficient):

VT b_Ax = P.get_mat()*p.getCoefficients();
NT* b_Ax_data() = b_Ax.data();

for (int = 0; i < m; i++) {  // then dist = *b_Ax_data;